spec/loophp/collection/CollectionSpec.php

```diff
@@ -305,9 +305,39 @@ public function it_can_be_constructed_from_a_generator(): void
             yield 'e';
         };
 
-        $this::fromIterable($generator())
-            ->getIterator()
+        $subject = $this::fromGenerator($generator());
+
+        $subject
             ->shouldIterateAs(range('a', 'e'));
+
+        $subject
+            ->count()
+            ->shouldBeEqualTo($subject->count());
+
+        $generator = static function () {
+            yield 'a';
+
+            yield 'b';
+
+            yield 'c';
+
+            yield 'd';
+
+            yield 'e';
+        };
+
+        $generator = $generator();
+        $generator->next();
+        $generator->next();
+
+        $subject = $this::fromGenerator($generator);
+
+        $subject
+            ->shouldIterateAs([2 => 'c', 3 => 'd', 4 => 'e']);
```

Comment:
is it possible to have a test where we do the same thing but with `fromIterable` so we can see what the difference is (if any) ?

Reply:
I'm guessing `fromIterable` will attempt to rewind the Generator and it fails actually, right?

Reply:
Job done!